### DISCUSSION TITLE
Sunton ESP32-S3 HMI 800x480 IPS Cap touch

### DISCUSSION THREAD

**Discussion by nikthefix** [moononournation/Arduino_GFX#225] [General]
https://www.aliexpress.com/item/1005004788147691.html

I just got this new board and combined with your terrific library it's awesome!

**Comment by nikthefix** [comment]
Please consider adding support for RM68120 display. Or is it the same as NT35510? I see a few very similar 800x480 displays with either controller.

**Reply by moononournation** [reply]
no in hand

**Comment by PowerTcX** [comment]
Hi, i also got this display, working great.
How to set display rotation for RGBP ?

```
Arduino_ESP32RGBPanel *bus = new Arduino_ESP32RGBPanel(
    GFX_NOT_DEFINED /* CS */, GFX_NOT_DEFINED /* SCK */, GFX_NOT_DEFINED /* SDA */,
    40 /* DE */, 41 /* VSYNC */, 39 /* HSYNC */, 42 /* PCLK */,
    45 /* R0 */, 48 /* R1 */, 47 /* R2 */, 21 /* R3 */, 14 /* R4 */,
    5 /* G0 */, 6 /* G1 */, 7 /* G2 */, 15 /* G3 */, 16 /* G4 */, 4 /* G5 */,
    8 /* B0 */, 3 /* B1 */, 46 /* B2 */, 9 /* B3 */, 1 /* B4 */
);
// option 1:
// ST7262 IPS LCD 800x480
 Arduino_RPi_DPI_RGBPanel *gfx = new Arduino_RPi_DPI_RGBPanel(
   bus,
   800 /* width */, 0 /* hsync_polarity */, 8 /* hsync_front_porch */, 4 /* hsync_pulse_width */, 8 /* hsync_back_porch */,
   480 /* height */, 0 /* vsync_polarity */, 8 /* vsync_front_porch */, 4 /* vsync_pulse_width */, 8 /* vsync_back_porch */,
   1 /* pclk_active_neg */, 14000000 /* prefer_speed */, true /* auto_flush */);
#endif /* !defined(DISPLAY_DEV_KIT) */ 
```

**Reply by moononournation** [reply]
not yet support

**Comment by nikthefix** [comment]
For those interested in this board you should be aware:
- great value for money and very good quality (makerfabs have a wiki)
- the native USB pins are not available without reworking the board.
- the touch panel interrupt pin is not connected as the 0ohm resistor shown
in the schematic connecting int>gpio18 is not fitted. A solder bridge will
do the job but board rework is then required as gpio18 is shared.

**Comment by Evohl** [comment]
Compilation error: no matching function for call to 'Arduino_ESP32RGBPanel::Arduino_ESP32RGBPanel(int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int, int)'

since I updated the library. works fine with older version (E.g. 1.2.8) but not anymore with newer versions :-(

**Reply by moononournation** [reply]
Please refer latest PDQgraphicstest for the new method signature.

**Reply by Evohl** [reply]
Got it working, very nice.
might be good to update the wiki
https://github.com/moononournation/Arduino_GFX/wiki/Dev-Device-Declaration
thanks a lot.

**Reply by moononournation** [reply]
oh yes, updated.

**Comment by volglizolic** [comment]
Hello!

I am using the above configs (the updated version) and I get an out of memory error from calling esp_lcd_new_rgb_panel..
Any ideas?

800x480 pixels

**Reply by moononournation** [reply]
Seems not yet enable PSRAM

**Comment by 1pYDca4d5TUYfb** [comment]
I've tried PDQgrapicstest for this board which seems to be running.  The serial output reports the timing for each test, but the screen does not actually display the tests.  the screen just cycles through various solid colors.  I also tried a very minimalist program that just picked random colors to fill the screen.  again the serial monitor reports that code is functioning as expected but the screen just cycles though the same basic colors.  any ideas on what might be wrong?

**Reply by nikthefix** [reply]
If I remember correctly, some of the variants of this display (and some others) have a built in self test which will cycle full screen colours if power is applied but the display is not initialised.  Maybe try uploading some code which is meant for a completely different purpose - blink sketch for example - and if you get the same colour cycles then that's what's going on. PDQ worked on this board for me but I had to be careful with my SPI and RGB data rate settings. If set too fast then the display would not init.

**Reply by 1pYDca4d5TUYfb** [reply]
thank you for your response. I'm pretty new to this so it can be tough to diagnose problems.  Just to confirm my understanding, by SPI data rate settings you mean using a non-default value in gfx->begin()? Is the rgb data rate the final parameter of the Arduino_ESP32RGBPanel constructor?  

Based on your suggestion I changed my code to gfx->begin(20000000) and the screen behaved as expected.  But then I changed it back to the default gfx->begin() and it still works.  so now I don't know if changing the spi frequency somehow reset the board/screen of if this is just some intermittent problem that for whatever reason is not happening right now.

**Reply by nikthefix** [reply]
Did you remove power to the board after changing settings? Without a power cycle the display init parameters left unaddressed will persist. 
Yes I meant non- default gfx-spi settings. I have this dev board so if you wish to share your code I'll gladly double test if you run in to further problems. Did you conclude that the colour cycle was indeed a test mode?

Bear in mind that the spi comms are for display setup. The parallel RGB data rate is set separately but from what you say this is working ok and hasn't been changed from library defaults.

**Reply by 1pYDca4d5TUYfb** [reply]
I appreciate the offer, but just your suggestion about the spi data rate was a huge help.  I'm trying to come up with a repeatable test case that results in the color cycling.  Over the last couple of weeks I tried so many different examples and the best I can tell is when I flash with a non-working example such as using incorrect board settings the board seems to stay in the test mode state even after loading working code.  changing the spi data rate to a lower than default value seems to reset screen.  As you noted, this might be a result of failing to power cycle between tests.

One remaining point of confusion for me is why the call to gfx->begin() is returning true when the screen is in this test mode.  I had this call wrapped in a while loop with a delay until it returns true so I was lead to believe that the screen was properly initialized.  Do you know of any way in code to test if the screen is in this test mode state?

**Reply by nikthefix** [reply]
The gfx->begin() will return true if the init code exectutes correctly and only when it has completed. There is no feedback from the display hardware to acknowledge that it initialized correctly unless you implement a display register read for confirmation. Arduino_GFX does not support display register read at the time of writing. LovyanGFX does. For testing purposes you could interogate some of the display registers manually via SPI. An oscilloscope is handy when investigating SPI signal integrity and working out the margins for a particular board. It's also woth picking up a couple of other boards which use the same processor and display type - Makerfabs and Lilygo for example. Different board layouts may give different results. I find the Lilygo boards particularly well designed.

**Comment by nielseng2017** [comment]
Recent purchase of "( From AliExprss: "ESP32-S3 HMI 8M PSRAM 16M Flash Arduino LVGL WIFI&Bluetooth 7" 800x480 Smart Display Screen 7.0 inch RGB LCD TFT module - Capacitive Touch" - which uses the JC8048B070N TFT LCD module.).
 
At power up the module ran its installed demo application just fine - looked great.

I followed the "Getting Started" instructions provided with the module:
Arduino IDE version 1.8.19 installed.
TFT_eSPI.h version  2.5.31 installed.
package_esp32_dev_index.json
Arduino_GFX_Library.h version 1.3.8

PROBLEM: I have not been able to run any Example or demo programs. Most fail compile with numerous errors.  The "Clock" demo compiles but nothing is displayed.

I'm most likely overlooking something obvious... 

Can you point me at a complete application/demo that is known to compile and run that was built using the Arduino IDE?

Any help you can provide would be greatly appreciated.

-George

**Reply by nikthefix** [reply]
Hi George. Can you elaborate? What is the nature of the errors. Arduino ESP32 recently updated to fall inline with the latest IDF version 5.xx

This has left older demo code broken in some cases - but not beyond repair. If in doubt ensure that versions of Arduino ESP32 / LVGL / TFT_eSPI / Arduino_GFX are as specified for the demo code. Then upgrade each component (starting with ESP32 boards) from there. The required changes are small - just nomenclature in most cases. Yes it's frustrating to have demo code for a new board which you find is out of date with respect to the root libraries. In my experience there is nearly always work to be done. Espressif have documented breaking changes as they relate to the upgrade from 4.xx to 5.xx

If your code compiles but doesn't produce results then see the above discussion about data rates.

**Comment by ZinggJM** [comment]
@moononournation, thank you very much for this excellent and versatile library!

I successfully tested with this display. Selected board ESP32S3 DevModule in Arduino IDE, with PSRAM: OPI PSRAM.

I reported some results in https://forum.arduino.cc/t/lvgl-desperation/1187605/9.

May I suggest to add board selection and specific board parameter(s) to the wiki Dev Device Declaration, as comment?

**Reply by nikthefix** [reply]
More than that, version control is a problem for many. All library versions need to be specified. After the IDF update 5.xx many older demo code is broken - but easily fixed by downgrading as long as we know the version numbers.

Perfect for testing is the old Arduino IDE portable option - create isolated IDE's for known working projects. Unfortunately this is not offered in the new 2.x IDE so version control is horrible.

We need to push for sketch based loading of IDE board settings - including lib versions. It's getting way too messy to have to do everything manually as the scope expands.

**Comment by Evohl** [comment]
For all that uses this Display I have something for you :) I designed a wall mount case for 3D Printing. Not sure if here is a good place to put the link but anyway there you go:
https://www.thingiverse.com/thing:6318814

**Reply by nikthefix** [reply]
Very nice! Can I suggest that you include the STEP file in your downloadables? Just so it's easy to add parametric custom mods. Unless of course you wish to produce and sell the enclosure in which case of course you should protect your IP.

**Reply by Evohl** [reply]
Did it with Tinkercad, seems there is no STEP file for download, sorry
